### PR TITLE
Upgrade to eslint 6.x and above

### PR DESCRIPTION
**Pull request by chrean**
Starting with eslint 6.x, `formatters` have been moved from `eslint/lib/formatters` into `eslint/lib/cli-engine/formatters`.

This fix will allow us to finally update all of our repositories to eslint 6.x or 7.x, whereas now we are stuck to 5.x.

Steps to test:

You can reproduce by using a repo that has this package as dependency.
By running `eslint -f json . | eslines` on a clone of that repo, you should get an error like the following:

```
node_modules/eslines/src/cli.js:44
				throw ex;
				^

Error: Problem loading formatter: eslint/lib/formatters/stylish
Error: Cannot find module 'eslint/lib/formatters/stylish'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
[CUT]
```

To see it fixed:

- Pull down this PR in a separate directory, say `~/eslines`.
- `cd ~/eslines` (or whatever your dir is)
- `npm pack` => will create `eslines-1.2.0.tgz`
- Go back to your "broken" repo and `npm I ~/eslines/eslines-1.2.0.tgz`

Tests should now be successful.

** Notes

Bumping the package version to 1.2.0 so anyone that is pinned to 1.1.0 won't break.

### CODE REVIEW

**Comment by chrean** [comment]
Closing as it seems to be not a priority plus it may some more work on `eslint-config-wpvip` which has related peer dependencies.

**Comment by nosolosw** [comment]
Happy to help test/support this change when you're ready.

**Comment by nosolosw** [comment]
:wave: @chrean I've updated the dev&prod dependencies as a preparation for this in a separate PR so will need to be rebased. Things that we need to do:

- package.json: add a new `peerDependencies` section where we list this requires eslint 6 or above.
- package.json: bump the package version to 2.0.0.
- changelog: add a message saying this is a breaking change, etc.
- Test instructions: an you provide updated test instructions with a public repo? I'm struggling to find one.

**Comment by nosolosw** [comment]
this fixes https://github.com/Automattic/eslines/issues/18

**Comment by chrean** [comment]
Hey @nosolosw ,

thanks for the feedback and the help!

I have made changes to CHANGELOG.md and package.json to reflect your suggestions, would like a sanity check though.

Will add instructions on how to test on public repos.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,17 @@
 # Changelog
 
 ## Next version
-[v1.1.0...master](https://github.com/Automattic/eslines/compare/v1.1.0...master)
+[v2.0.0...master](https://github.com/Automattic/eslines/compare/v2.0.0...master)
 
+## 2020-05-19 v2.0.0
+[v1.1.0...v2.0.0](https://github.com/Automattic/eslines/compare/v1.1.0...v2.0.0)
+
+#### Warning: this release breaks backward compatibility!
```
Comment (nosolosw) [inline CHANGELOG.md:9]:
```suggestion
#### Breaking change
```

Comment (nosolosw) [inline CHANGELOG.md:9]: How about this?

```
- Bump dev & prod dependencies to the latest versions.

### Breaking changes

- eslines now requires a peer dependency of ESLint 6 or above. [#20](https://github.com/Automattic/eslines/pull/20)
```

File: package.json [hunk 2]
```diff
@@ -32,11 +32,14 @@
   },
   "devDependencies": {
     "babel-eslint": "10.1.0",
-    "eslint": "7.0.0",
+    "eslint": ">=6.0.0",
```
Comment (nosolosw) [inline package.json:35]: does this need to be in sync with the peer dependencies, or can we use here the latest ESLint?

**Comment by chrean** [comment]
Thanks @nosolosw for suggestions, both make sense.
Note for the future: how about using GitHub suggestions so we can easily commit changes?

File: package.json (hunk 2 above)
Comment (chrean) [reply package.json:35]: This was already fixed, perhaps you missed a refresh, good point anyway! 🖖

Comment (nosolosw) [reply package.json:35]: I'm still seeing the 6 or later :thinking: See capture below:

Screenshot from 2020-05-19 14-15-26

Comment (chrean) [reply package.json]: Oh sorry I missed your point totally 🤦‍♂️
Fixed now with 2679afe

File: CHANGELOG.md (hunk 1 above)
Comment (chrean) [reply CHANGELOG.md]: Makes sense, fixed here: afacc12

**Review by nejclovrencic — APPROVED** [review]

**Pull request merged**